File: day1/cxx/puzzle2.cxx
#include <algorithm>
#include <fstream>
#include <iostream>
#include <numeric>
#include <string>
#include <vector>

#include "lines.hpp"

using namespace std;

const uint32_t solve()
{
    vector<uint32_t> calories = {0};

    ifstream f{"../input.txt"};

    for (auto & line: lines(f))
    {
        if (line.empty())
            calories.push_back(0);
        else
            calories.back() += atoi(line.c_str());
    }

    sort(calories.begin(), calories.end());

    return accumulate(calories.end()-3, calories.end(), 0u);
}

int main()
{
    auto solution = solve();
    cout << solution << endl;
}

File: day1/py/puzzle2.py
#!/usr/bin/env python

def solve():
    calories = [0]

    with open('../input.txt', 'r') as f:
        for line in f:
            if line == "\n":
                calories.append(0)
            else:
                calories[-1] += int(line)

    c = sorted(calories)

    return c[-1] + c[-2] + c[-3] 

if __name__ == "__main__":
    solution = solve()
    print(solution, )

